### PR TITLE
Update Trace API for 0.6

### PR DESCRIPTION
**Pull request by omerfirmak**

### CODE REVIEW

File: vm/rust/src/jsonrpc.rs [hunk 1]
```diff
@@ -203,18 +244,15 @@ pub struct FunctionInvocation {
     pub calls: Vec<FunctionInvocation>,
     pub events: Vec<OrderedEvent>,
     pub messages: Vec<OrderedMessage>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub execution_resources: Option<ExecutionResources>,
```
Comment (omerfirmak) [inline vm/rust/src/jsonrpc.rs:248]: We are missing this when we adapt traces from the feeder. Do you mind adding that on top of this PR @joshklop?

**Review by joshklop — APPROVED** [review]

File: vm/rust/src/jsonrpc.rs (hunk 1 above)
Comment (joshklop) [reply vm/rust/src/jsonrpc.rs:248]: Will do.

File: vm/rust/src/jsonrpc.rs [hunk 2]
```diff
@@ -185,12 +185,53 @@ use blockifier::execution::call_info::OrderedEvent as BlockifierOrderedEvent;
...
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub ecdsa_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub bitwise_builtin_applications: Option<usize>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub keccak_builtin_applications: Option<usize>,
+    // https://github.com/starkware-libs/starknet-specs/pull/167
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub segment_arena_builtin: Option<usize>,
+}
```
Comment (omerfirmak) [inline vm/rust/src/jsonrpc.rs:215]: Also, these must be marshaled in to a hex string if json is legacy :/

Comment (joshklop) [reply vm/rust/src/jsonrpc.rs:215]: That's a nice catch

Comment (omerfirmak) [reply vm/rust/src/jsonrpc.rs]: I think I got it confused sorry:/ Trace API v0.5 doesnt have execution resources, so there is no need to have the ability to marshal these into hex strings.

But we do need to support that for receipts.

**Pull request merged**